FitGoal/FitGoal/Singup&Singin/Authenticator.swift

```diff
@@ -41,13 +41,11 @@ class Authenticator: NSObject, GIDSignInDelegate {
             }
         }
         
-        switch self.socialMedia {
-        case .facebook:
-            facebookSignIn(sender: sender, completion: persistIfPossible(userInfoResult:))
-        case .google:
-            googleSignIn(sender: sender, completion: persistIfPossible(userInfoResult:))
-        case .twitter:
-            twitterSignIn(sender: sender, completion: persistIfPossible(userInfoResult:))
+        switch authenticationMethod {
+        case .custom(let customAuth):
+            customSignIn(customAuth: customAuth, completion: persistIfPossible(userInfoResult:))
+        case .socialMedia(let socialMedia):
+            authenticateWithSocialMedia(socialMedia: socialMedia, sender: sender, completion: persistIfPossible(userInfoResult:))
```

Comment:
The name of this function can be improved. Now `socialMedia` is a suffix of the function name and also the external name of the first parameter.